samples/storage/src/main/java/com/example/platform/storage/mediastore/MediaStoreQuery.kt

```diff
@@ -0,0 +1,169 @@
+    description = "Query files indexed by MediaStore",
+    documentation = "https://developer.android.com/training/data-storage/shared/media#media_store",
+)
+@SuppressLint("MissingPermission")
+@Composable
+fun MediaStoreQueryScreen() {
+    val permission = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        READ_MEDIA_IMAGES
+    } else {
+        READ_EXTERNAL_STORAGE
+    }
+
+    PermissionBox(
+        permission = permission,
+        onGranted = { MediaStoreQueryContent() },
+        description = "Permission needed to access the device's photo library",
+    )
+}
+
+@RequiresPermission(anyOf = [READ_MEDIA_IMAGES, READ_EXTERNAL_STORAGE])
+@Composable
+fun MediaStoreQueryContent() {
+    val context = LocalContext.current
+    var files by remember { mutableStateOf(emptyList<FileEntry>()) }
+
+    LaunchedEffect(Unit) {
+        files = getImages(context.contentResolver)
+    }
+
+    Column(Modifier.fillMaxSize()) {
+        LazyColumn(Modifier.weight(1f)) {
+            item {
+                ListItem(
+                    headlineContent = {
+                        if (files.isNotEmpty()) {
+                            Text("${files.size} images found")
+                        }
+                    },
+                )
+                Divider()
+            }
+            items(files) { file ->
+                ListItem(
+                    leadingContent = {
+                        AsyncImage(
+                            model = file.uri,
+                            contentDescription = null,
+                            contentScale = ContentScale.Crop,
+                            modifier = Modifier
+                                .size(64.dp)
+                                .aspectRatio(1f),
+                        )
+                    },
+                    headlineContent = {
+                        Text(file.name, maxLines = 1, overflow = TextOverflow.Ellipsis)
+                    },
+                    supportingContent = { Text(file.mimeType) },
+                    trailingContent = { Text(Formatter.formatShortFileSize(context, file.size)) },
+                )
+                Divider()
+            }
+        }
+    }
+}
+
+suspend fun getImages(contentResolver: ContentResolver): List<FileEntry> {
+    return withContext(Dispatchers.IO) {
+        val projection = arrayOf(
+            Images.Media._ID,
+            Images.Media.DISPLAY_NAME,
+            Images.Media.SIZE,
+            Images.Media.MIME_TYPE,
+            Images.Media.DATE_ADDED,
+        )
+
+        val collectionUri = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            // This allows us to query all the device storage volumes instead of the primary only
+            Images.Media.getContentUri(MediaStore.VOLUME_EXTERNAL)
+        } else {
+            Images.Media.EXTERNAL_CONTENT_URI
+        }
+
+        val images = mutableListOf<FileEntry>()
+
+        contentResolver.query(
```

Comment:
For the sample readability can you add params name and explain why the null values

Reply:
Named arguments are not allowed for non-Kotlin functions. I've added inline comments

Reply:
Yup, that's what I meant.